homeassistant/components/palazzetti/sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Support for Palazzetti sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfMass, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import PalazzettiConfigEntry
+from .coordinator import PalazzettiDataUpdateCoordinator
+from .entity import PalazzettiEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class CallableSensorEntityDescription(SensorEntityDescription):
+    """Describes a Palazzetti sensor entity that is read from a `Callable`."""
+
+    value_callable: Callable[[], int | float | str]
```

Comment:
I'm also thinking, since you check for isinstance anyways might set this to `None` by default, unify the two descriptions and check for `None` during native_value?

Reply:
Callable Sensors are based on a value that is returned by a callable. They don't need a presence flag because the API returns a list of only the ones that are present on a given device model. This is only the case for temperature sensors at the moment, because the number of sensors and their named vary between models.